node/src/BaseClient.ts

```diff
@@ -1183,18 +1183,21 @@ export class BaseClient {
      * See https://redis.io/commands/smembers/ for details.
      *
      * @param key - The key to return its members.
-     * @returns All members of the set.
-     * If `key` does not exist, it is treated as an empty set and this command returns empty list.
+     * @returns A set containing all members of the set.
+     * If `key` does not exist, it is treated as an empty set and this command returns an empty set.
      *
      * @example
      * ```typescript
      * // Example usage of the smembers method
      * const result = await client.smembers("my_set");
-     * console.log(result); // Output: ["member1", "member2", "member3"]
+     * console.log(result); // Output: Set {'member1', 'member2', 'member3'}
      * ```
      */
-    public smembers(key: string): Promise<string[]> {
-        return this.createWritePromise(createSMembers(key));
+    public async smembers(key: string): Promise<Set<string>> {
+        const result = (await this.createWritePromise(
+            createSMembers(key),
+        )) as string[];
+        return new Set(result);
```

Comment:
A breaking change for user API
I think we have to document it in changelog and/or release notes.

Reply:
I documented that in the changelog

Reply:
There is a good practice to highlight those as a breaking change and/or track these changes in another doc
Should be also highlighted in release notes